modules/imgcodecs/src/loadsave.cpp

```diff
@@ -284,6 +306,50 @@ static void ApplyExifOrientation(const String& filename, Mat& img)
     }
 }
 
+static void ApplyExifOrientation(const String& filename, Mat& img)
+{
+    int orientation = IMAGE_ORIENTATION_TL;
+
+    if (filename.size() > 0)
+    {
+       std::ifstream stream( filename.c_str(), std::ios_base::in | std::ios_base::binary );
+       ExifReader reader( stream );
+       if( reader.parse() )
+       {
+           ExifEntry_t entry = reader.getTag( ORIENTATION );
+           if (entry.tag != INVALID_TAG)
+           {
+               orientation = entry.field_u16; //orientation is unsigned short, so check field_u16
+           }
+       }
+       stream.close();
+    }
+
+    ExifTransform(orientation, img);
+}
+
+static void ApplyExifOrientation(const Mat& buf, Mat& img)
+{
+    int orientation = IMAGE_ORIENTATION_TL;
+
+    if( buf.isContinuous() )
+    {
+       ByteStreamBuffer bsb( reinterpret_cast<char*>(buf.data), buf.total() * buf.elemSize() );
+       std::istream stream( &bsb );
```

Comment:
i'm not sure this is what caused build to break - it passed with a nearly identical implementation